src/dbnode/persist/fs/seek_manager.go

```diff
@@ -498,6 +504,68 @@ func (m *seekerManager) updateOpenLeaseHotSwapSeekers(
 	return wg, updateOpenLeaseResult, nil
 }
 
+func (m *seekerManager) RelinquishShard(shard uint32) error {
+	byTime := m.seekersByTime(shard)
+
+	// Indicate this shard is closed so it can't be borrowed and no seekers
+	// can be opened from now on.
+	byTime.Lock()
+	byTime.closed = true
+	byTime.Unlock()
+
+	for _, seekers := range byTime.seekers {
+		for _, s := range []seekersAndBloom{seekers.active, seekers.inactive} {
+			anySeekersAreBorrowed := false
+			for _, seeker := range s.seekers {
+				if seeker.isBorrowed {
+					anySeekersAreBorrowed = true
+					break
+				}
+			}
+
+			if anySeekersAreBorrowed {
+				// If any of the seekers are borrowed setup a waitgroup which will be used to
+				// signal when they've all been returned (the last seeker that is returned via
+				// the Return() API will call wg.Done()).
+				byTime.Lock()
+				if s.wg != nil {
+					wg := &sync.WaitGroup{}
+					wg.Add(1)
+					s.wg = wg
+				} else {
+					s.wg.Add(1)
```

Comment:
I don't think you want to add 1 because then done will need to be called twice for it to be released but it will only be called once.

You have have multiple waiters for the same waitgroup even though Add() has only been called once so I think you can just remove this line

Reply:
Ok yeah that makes sense.